library/src/main/java/pl/rspective/voucherify/android/view/OnValidatedListener.java

```diff
@@ -0,0 +1,11 @@
+package pl.rspective.voucherify.android.view;
+
+import pl.rspective.voucherify.android.client.model.VoucherResponse;
+import retrofit.RetrofitError;
+
+public interface OnValidatedListener {
+
+    void onValidated(VoucherResponse result);
```

Comment:
Right. Need to add some docs. Currently `onValidated` is called in both cases - when voucher code is valid or not - you can distinguish by checking VoucherResponse.isValid(). The `onError` is called when some a serious error occurred - e.g. invalid API keys.